### PR TITLE
Remove Istiod MCP sink

### PR DESCRIPTION
**Pull request by howardjohn**
This has been deprecated in favor of the XDS-over-MCP sink, which has been informally announced to users for many months. In networking WG today we agreed on its removal for 1.9. We will add an additional warning in the 1.8 notes to ensure people have at least 4 months notice of this change.

### CODE REVIEW

**Review by nmittler — APPROVED** [review]

**Review by jasonwzm — APPROVED** [review]

**Review by sdake — APPROVED** [review]
@howardjohn this must have taken a couple red bulls to get through :) I think this removal is fine. It doesn't follow our deprecation policy (we  deprecate first, then remove second) - although I don't know of anyone using MCP. Although just because I don't know its happening, doesn't mean individuals have become dependent on this feature.

Still, approval looks good since this code is not really used that we are aware of..

I would encourage you to send a mail to discuss.istio.io to communicate MCP is being removed in 1.9 and offer inks to MCP over XDS APIs so anyone that is using this functionality has an opportunity to protest. (same with a slack announcement).

Your call.

Cheers,
-steve

File: releasenotes/notes/mcp-removal.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+releaseNotes:
+- |
+  **Removed** support for reading Istio configuration over the Mesh Configuration Protocol (MCP).
```
Comment (costinm) [inline releasenotes/notes/mcp-removal.yaml:6]: Should we mention the config-over-XDS ( like to design doc ) ? It is experimental, but it's good to indicate a replacement may be possible if anyone depends on the removed feature.

**Review by costinm — APPROVED** [review]

**Comment by davidfleming-keyfactor** [comment]
Hi @costinm @howardjohn 

The removal breaks the following PR -> https://github.com/istio/istio/pull/26316

See -> https://github.com/istio/istio/pull/26316/files#diff-c2aac71be81e1ae63701f87003f4ec178eb22d708ecbb8ab123128aa6971dbe6R30

Can you please assist or provide documentation for switching over to XDS-over-MCP/config-over-XDS?

Thanks,
David

**Pull request merged**